src/EnergyPlus/OutputReportTabular.cc

```diff
@@ -13254,32 +13238,34 @@ namespace OutputReportTabular {
                 for (int jSurf = Zone(zoneIndex).SurfaceFirst; jSurf <= Zone(zoneIndex).SurfaceLast; ++jSurf) {
                     if (!Surface(jSurf).HeatTransSurf) continue; // Skip non-heat transfer surfaces
 
-                    // determine for each timestep the amount of radiant heat for each end use absorbed in each surface
-                    Real64 QRadThermInAbsMult =
-                        TMULTseq(desDaySelected, kTimeStep, radEnclosureNum) * ITABSFseq(desDaySelected, kTimeStep, jSurf) * Surface(jSurf).Area;
-                    peopleRadIntoSurf(kTimeStep) = peopleRadSeq(desDaySelected, kTimeStep, zoneIndex) * QRadThermInAbsMult;
-                    equipRadIntoSurf(kTimeStep) = equipRadSeq(desDaySelected, kTimeStep, zoneIndex) * QRadThermInAbsMult;
-                    hvacLossRadIntoSurf(kTimeStep) = hvacLossRadSeq(desDaySelected, kTimeStep, zoneIndex) * QRadThermInAbsMult;
-                    powerGenRadIntoSurf(kTimeStep) = powerGenRadSeq(desDaySelected, kTimeStep, zoneIndex) * QRadThermInAbsMult;
-                    lightLWRadIntoSurf(kTimeStep) = lightLWRadSeq(desDaySelected, kTimeStep, zoneIndex) * QRadThermInAbsMult;
-                    // for each time step, step back through time and apply decay curve
+                    // for each time step, step back through time and apply decay curve to radiant heat for each end use absorbed in each surface
                     Real64 peopleConvFromSurf = 0.0;
                     Real64 equipConvFromSurf = 0.0;
                     Real64 hvacLossConvFromSurf = 0.0;
                     Real64 powerGenConvFromSurf = 0.0;
                     Real64 lightLWConvFromSurf = 0.0;
                     Real64 lightSWConvFromSurf = 0.0;
                     Real64 feneSolarConvFromSurf = 0.0;
-                    for (int mStepBack = 1; mStepBack <= kTimeStep; ++mStepBack) {
-                        peopleConvFromSurf += peopleRadIntoSurf(kTimeStep - mStepBack + 1) * decayCurve(mStepBack, jSurf);
-                        equipConvFromSurf += equipRadIntoSurf(kTimeStep - mStepBack + 1) * decayCurve(mStepBack, jSurf);
-                        hvacLossConvFromSurf += hvacLossRadIntoSurf(kTimeStep - mStepBack + 1) * decayCurve(mStepBack, jSurf);
-                        powerGenConvFromSurf += powerGenRadIntoSurf(kTimeStep - mStepBack + 1) * decayCurve(mStepBack, jSurf);
-                        lightLWConvFromSurf += lightLWRadIntoSurf(kTimeStep - mStepBack + 1) * decayCurve(mStepBack, jSurf);
+
+                    for (int mStepBack = 1; mStepBack <= NumOfTimeStepInHour * 24; ++mStepBack) {
+                        int sourceStep = kTimeStep - mStepBack + 1;
+                        if (mStepBack > kTimeStep) {
+                            sourceStep = NumOfTimeStepInHour * 24 - (mStepBack - kTimeStep) + 1;
+                        }
+                        assert(sourceStep > 0);
```

Comment:
Now the history is accumulated for a full 24 hours of the selected day. For 5ZoneAirCooled this made very tiny differences in the component loads, but for very heavy constructions or peaks early in the day, it could matter more.

Reply:
These time series are hard to align. And hard to understand what time is referred to with each. If I understand this then decayCurve(1) starts at the time of the pulse (largest fraction) and decays over 24 hours. kTimeStep is 1 at the beginning of the day (hr 00:015). mStepBack is 1 at the time of pulse. This needs to be understood to understand the math for source step.

What I am looking at is the decayCurve time series and it looks like it's not a full day if the pulse happens during the day. Should this loop to TimeOfPulse + NumOfTimeStepInHour * 24? to get a full day? And the diff equation would then need to account for midnight to get back to index 1 so the array size isn't exceeded. Same for decayCurveHeat.

line 12347 of OutputReportTabular

                for (TimeStep = TimeOfPulse; TimeStep <= NumOfTimeStepInHour * 24; ++TimeStep) {
                    if (radiantPulseReceived(CoolDesSelected, SurfNum) != 0.0) {
                        diff = loadConvectedWithPulse(CoolDesSelected, TimeStep, SurfNum) - loadConvectedNormal(CoolDesSelected, TimeStep, SurfNum);
                        decayCurveCool(TimeStep - TimeOfPulse + 1, SurfNum) = -diff / radiantPulseReceived(CoolDesSelected, SurfNum);
                    } else {
                        decayCurveCool(TimeStep - TimeOfPulse + 1, SurfNum) = 0.0;
                    }
                }

Reply:
I'd also like to know what the sum of the decayCurve fractions are.

Reply:
So the pulse always happens on hour = 10 in bool CalcdoLoadComponentPulseNow. That means the decay is only for hour 10 to 24,

Reply:
The docs seems to say that this pulse run is supposed to be applied to the peak load based on the averaging window. Also a mistake in wording in the 2nd to last sentence (i.e., before is used twice). This model seems to be doing this but I still have questions, namely the time of pulse will influence the decay curve fraction for the peak hour, the docs also mention this.

    Using the internal and solar gain results saved from the normal sizing period in step (2),
    for each timestep prior to and including the time of the peak load during the sizing day,
    the decay curve is applied to each radiant gain on each surface for each timestep to
    generate a predicted delayed load component for the internal and solar gains for each
    timestep that comprise the peak load (based on Timesteps in Averaging Window). 
    Timesteps just before the peak have much larger impacts than those just a few timesteps
    before the peak. These results will be the radiant portion of the load for each type of
    internal and solar gain.

Reply:
So the numbers shown in the zone component load summary are not a summation of the decay over time but the impact of radiant to convection heat during the peak load time. So what happens if the peak load occurs at 9 AM when the pulse occurs at 10 AM.d This seems to imply that the decay curve needs to wrap around midnight. And then the decay curve fraction would be small at 9 AM and would not give the correct results for radiant to convection heat at 9 AM. I've only got questions instead of answers. I guess the fix proposed here does fix a problem so my concerns, if valid, would require a follow up issue, not hold up this one.

Reply:
Mike, I'm not sure I agree with wrapping around the design day as being correct. It is assuming implying that multiple (two) design days occur in a row. That seems unlikely to me. Isn't it more likely that a design day occurs after a fairly normal day? Given that, taking the impact from the end of the current design day as the delayed effect of the "previous" day seems to be overstating the loads.

Reply:
@rraustad The decay curve is not aligned to any specific time. DecayCurve(1:N) is the fraction of the pulse that becomes a convective gain from a given surface in the timesteps at and following the pulse. The series should sum to 1. This pattern (e.g. 0.2, 0.1, 0.05, 0.04 . . . ) is then applied to incident radiant fluxes on each surface. Applying the pulse at 10am is just a way to get the decay pattern, the same decay pattern is then applied to any incident radiant (and solar) flux at any time of day.

As for the timestep formulas, I had to make a spreadsheet to get those right. That said, adding the extension to 24 hrs seemed like a good idea, but I probably shouldn't have clouded this PR with that. Maybe I should revert the 24 hr part and just keep the other parts of the fix. Either that or I need to boost the comments and docs to better explain it.

Reply:
I think the assumption here is that the radiant loads would be similar on the previous day so using the design day like this seems OK to me. One could argue a cloudy prior day or a Monday peak with different schedules the previous day but for now this works well enough.

Reply:
After reading that sentence over and over again I guess it makes sense.

    Timesteps just before the peak have much larger impacts than those just a few timesteps
before the peak.